.github/workflows/dispatch-yocto-build-deploy.yml

```diff
@@ -0,0 +1,124 @@
+name: "Test Yocto"
+
+on:
+  pull_request:
+    types: [opened, synchronize]
+    branches:
+      - "main"
+      - "master"
+  pull_request_target:
+    types: [opened, synchronize]
+    branches:
+      - "main"
+      - "master"
+
+  workflow_dispatch:
+    # you may only define up to 10 `inputs` for a `workflow_dispatch` event
+    inputs:
+      device-repo:
+        description: balenaOS device repository (owner/repo)
```

Comment:
Thinking whether we should just place the build script in the device repository (maybe an include from a common location) so that this information is provided by github.

Reply:
This `dispatch-` workflow is an example of what will be in each device repo, and it is used to call the shared workflow like we do with Flowzone.

The device-repo field is only required when testing it from balena-yocto-scripts, and would not be exposed when this script is installed on device repos. In fact, on device repos we might only use the `pull_request` trigger and not allow manual execution via `workflow_dispatch` at all, instead using this copy in balena-yocto-scripts for manual execution and testing.